### DISCUSSION TITLE
Eval frequency in epochs?

### DISCUSSION THREAD

**Discussion by dakinggg** [explosion/spaCy#6656] [Help: Coding & Implementations]
It looks like the `eval_frequency` param to training is in steps. Rather than evaluating every number of steps, I'd like to evaluate each epoch. Any plans to include this, or perhaps I'm missing how to do it already? The number of steps per epoch could be calculated, except that we are using a compounding batch size. All of this applies to the `patience` param as well. Thanks!

**Comment by honnibal** [comment]
This is one of those options that got pruned away. Apart from exactly mimicking another system (or version), is there any reason why always evaluating at the end of an epoch is better than evaluating at some arbitrary number of examples?

I used the number of steps instead because if you have a large dataset, it can be annoying to not evaluate often enough. I even considered some crazy setup with wall clock time, where you say "spend 5% of the training time on evaluation", but I decided that was too whacky.

**Reply by dakinggg** [reply]
It was mostly just about watching the model train to make sure I haven't horribly screwed something up :) Using steps makes a lot of sense for larger datasets imo. For tracking progress, it is nice to know how many epochs it has gone through. e.g. I just accidentally set the eval_frequency way too high, and so it seemed like training was moving at a glacial pace relative to how long I remember these models taking to train, but really it had gone through multiple epochs without printing any progress. Since then I discovered the `progress_bar` param, which at least tells me its not stuck. In summary, pretty minor, I'm just used to checking time and performance per epoch on these models to make sure they're training ok.
